common/djangoapps/student/helpers.py

```diff
@@ -247,7 +248,7 @@ def get_next_url_for_login_page(request):
 
     # if we get a redirect parameter, make sure it's safe. If it's not, drop the
     # parameter.
-    if redirect_to and not http.is_safe_url(redirect_to):
+    if redirect_to and (not http.is_safe_url(redirect_to) or settings.STATIC_URL in redirect_to):
```

Comment:
I think it's time to move the logic behind the first `and` into a new function and give it a better name.